kernel/page_allocator/src/lib.rs

```diff
@@ -624,6 +677,7 @@ fn adjust_chosen_chunk(
 pub fn allocate_pages_deferred(
 	requested_vaddr: Option<VirtualAddress>,
 	num_pages: usize,
+	within_range: Option<&PageRange>,
```

Comment:
Currently it's possible to request both a specific `vaddr` and a specific `within_range`.

May be better to do something like
```rust
pub enum MemoryRequest {
    Address(VirtualAddress),
    Range(PageRange),
}

pub fn allocate_pages_deferred(num_pages: usize, request: MemoryRequest);
```
(probably with a name better than `MemoryRequest`).

Reply:
ha, yea i have that lined up for a future PR already. Great minds think alike.